src/realm/query_expression.hpp

```diff
@@ -1540,56 +1619,64 @@ class Value : public ValueBase, public Subexpr2<T> {
             // linked-to-value fulfills the condition
             REALM_ASSERT_DEBUG(left->m_values > 0);
             const size_t num_right_values = right->m_values;
-            if (right_cmp_type == ExpressionComparisonType::Any) {
```

Comment:
I find this refactoring a bit harder to read, but that might be because of github's diff view 🤷‍♂️

Reply:
Yes - the structure is changed somewhat.